### PR TITLE
Checked for nullity of activation / deactivation results

### PR DESCRIPTION
**Pull request by ikbalkaya**
This PR  fixes https://github.com/ably/ably-flutter/issues/298

`resultForActivate` and `resultForDeactivate` will be checked before methods using them are invoked

### CODE REVIEW

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java [hunk 1]
```diff
@@ -64,13 +64,21 @@ public void onReceive(Context context, Intent intent) {
       switch (action) {
         case PUSH_ACTIVATE_ACTION:
           callCallbackOnDartSide(PlatformConstants.PlatformMethod.pushOnActivate, errorInfo);
-          returnMethodCallResult(resultForActivate, errorInfo);
-          resultForActivate = null;
+          if (resultForActivate != null){
+            returnMethodCallResult(resultForActivate, errorInfo);
+            resultForActivate = null;
+          }else{
+            Log.e(TAG, "resultForActivate is null: ", new NullPointerException());
```
Comment (QuintinWillison) [inline android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:71]:
```suggestion
            Log.e(TAG, "resultForActivate is null on PUSH_ACTIVATE_ACTION.");
```

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java [hunk 2]
```diff
@@ -64,13 +64,21 @@ public void onReceive(Context context, Intent intent) {
...
           break;
         case PUSH_DEACTIVATE_ACTION:
           callCallbackOnDartSide(PlatformConstants.PlatformMethod.pushOnDeactivate, errorInfo);
-          returnMethodCallResult(resultForDeactivate, errorInfo);
-          resultForDeactivate = null;
+          if (resultForDeactivate != null){
+            returnMethodCallResult(resultForActivate, errorInfo);
+            resultForDeactivate = null;
+          }else{
+            Log.e(TAG, "resultForDeactivate is null: ", new NullPointerException());
```
Comment (QuintinWillison) [inline android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:80]:
```suggestion
            Log.e(TAG, "resultForDeactivate is null on PUSH_DEACTIVATE_ACTION.");
```

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java [hunk 3]
```diff
@@ -64,13 +64,21 @@ public void onReceive(Context context, Intent intent) {
       switch (action) {
         case PUSH_ACTIVATE_ACTION:
           callCallbackOnDartSide(PlatformConstants.PlatformMethod.pushOnActivate, errorInfo);
-          returnMethodCallResult(resultForActivate, errorInfo);
-          resultForActivate = null;
+          if (resultForActivate != null){
+            returnMethodCallResult(resultForActivate, errorInfo);
```
Comment (QuintinWillison) [inline android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:68]: Let's add an affirming log spew above this also. DEBUG level.

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java [hunk 4]
```diff
@@ -64,13 +64,21 @@ public void onReceive(Context context, Intent intent) {
...
+          }else{
+            Log.e(TAG, "resultForActivate is null: ", new NullPointerException());
+          }
           break;
         case PUSH_DEACTIVATE_ACTION:
           callCallbackOnDartSide(PlatformConstants.PlatformMethod.pushOnDeactivate, errorInfo);
-          returnMethodCallResult(resultForDeactivate, errorInfo);
-          resultForDeactivate = null;
+          if (resultForDeactivate != null){
+            returnMethodCallResult(resultForActivate, errorInfo);
```
Comment (QuintinWillison) [inline android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:77]: Let's add an affirming log spew above this also. DEBUG level.

**Review by QuintinWillison — CHANGES_REQUESTED** [review]

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java (hunk 4 above)
Comment (KacperKluka) [inline android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:79]: I think you should use the `resultForDeactivate` in this method :wink: 
```suggestion
            returnMethodCallResult(resultForDeactivate, errorInfo);
```

**Review by KacperKluka — CHANGES_REQUESTED** [review]
One change required and please reformat the code as some whitespace are missing :wink:

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java (hunk 4 above)
Comment (QuintinWillison) [reply android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:79]: Well spotted! Can't believe I missed that on my first review pass. 😳

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java (hunk 3 above)
Comment (ikbalkaya) [reply android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:69]: https://github.com/ably/ably-flutter/pull/303/commits/0e8bacd633b39eb79ce08df9eb1c35cf19224e67

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java (hunk 4 above)
Comment (ikbalkaya) [reply android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:79]: https://github.com/ably/ably-flutter/pull/303/commits/0e8bacd633b39eb79ce08df9eb1c35cf19224e67

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java (hunk 1 above)
Comment (ikbalkaya) [reply android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java]: https://github.com/ably/ably-flutter/pull/303/commits/6040d1f6d8b11fe1d643f2412d2544dc11944e47

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java (hunk 2 above)
Comment (ikbalkaya) [reply android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java]: https://github.com/ably/ably-flutter/pull/303/commits/6040d1f6d8b11fe1d643f2412d2544dc11944e47

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java (hunk 4 above)
Comment (ikbalkaya) [reply android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:79]: Oops 😮

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java [hunk 5]
```diff
@@ -64,13 +64,23 @@ public void onReceive(Context context, Intent intent) {
       switch (action) {
         case PUSH_ACTIVATE_ACTION:
           callCallbackOnDartSide(PlatformConstants.PlatformMethod.pushOnActivate, errorInfo);
-          returnMethodCallResult(resultForActivate, errorInfo);
-          resultForActivate = null;
+          if (resultForActivate != null){
+            Log.d(TAG, "resultForActivate is null on PUSH_ACTIVATE_ACTION.");
```
Comment (QuintinWillison) [inline android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:68]:
```suggestion
            Log.d(TAG, "resultForActivate received on PUSH_ACTIVATE_ACTION.");
```

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java [hunk 6]
```diff
@@ -64,13 +64,23 @@ public void onReceive(Context context, Intent intent) {
...
+          }else{
+            Log.e(TAG, "resultForActivate is null on PUSH_ACTIVATE_ACTION.");
+          }
           break;
         case PUSH_DEACTIVATE_ACTION:
           callCallbackOnDartSide(PlatformConstants.PlatformMethod.pushOnDeactivate, errorInfo);
-          returnMethodCallResult(resultForDeactivate, errorInfo);
-          resultForDeactivate = null;
+          if (resultForDeactivate != null){
+            Log.d(TAG, "resultForDeactivate is null on PUSH_DEACTIVATE_ACTION.");
```
Comment (QuintinWillison) [inline android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:78]:
```suggestion
            Log.d(TAG, "resultForDeactivate received on PUSH_DEACTIVATE_ACTION.");
```

**Review by QuintinWillison — CHANGES_REQUESTED** [review]

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java (hunk 4 above)
Comment (ikbalkaya) [reply android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java]: Thanks for spotting this serious copy/paste error @KacperKluka . https://github.com/ably/ably-flutter/pull/303/commits/20cf0187930b52654aca95a5151e69b34b8b271c

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java [hunk 7]
```diff
@@ -64,13 +64,23 @@ public void onReceive(Context context, Intent intent) {
       switch (action) {
         case PUSH_ACTIVATE_ACTION:
           callCallbackOnDartSide(PlatformConstants.PlatformMethod.pushOnActivate, errorInfo);
-          returnMethodCallResult(resultForActivate, errorInfo);
-          resultForActivate = null;
+          if (resultForActivate != null){
+            Log.d(TAG, "resultForActivate received on PUSH_ACTIVATE_ACTION.");
+            returnMethodCallResult(resultForActivate, errorInfo);
+            resultForActivate = null;
+          }else{
```
Comment (KacperKluka) [inline android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:71]: Please run code auto-formatting as this line (and a few more) are missing whitespace :wink: 
```suggestion
          } else {
```

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java (hunk 5 above)
Comment (QuintinWillison) [reply android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java]: https://github.com/ably/ably-flutter/pull/303/commits/61e868d442c07ec8844f32a75d8ca1cb519a9bfa

**Review by KacperKluka — CHANGES_REQUESTED** [review]
Just a small cosmetic change :wink:

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java (hunk 3 above)
Comment (QuintinWillison) [reply android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:69]: https://github.com/ably/ably-flutter/pull/303/commits/61e868d442c07ec8844f32a75d8ca1cb519a9bfa

**Review by QuintinWillison — APPROVED** [review]

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java (hunk 7 above)
Comment (ikbalkaya) [reply android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:71]: Autoformatting will generally cause issues (lots of changes) on plugin code as there is no configuration (AFAIK) set up. I will try to manually format it

Comment (KacperKluka) [reply android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java]: Given there's no style configuration I wouldn't want to block you on manually formating the code :wink: Therefore, I'll resolve this comment and approve the PR as this is just a cosmetic change :+1:

**Review by KacperKluka — APPROVED** [review]
LGTM :+1:

**Pull request merged**